parsing/ast_helper.mli

```diff
@@ -73,6 +73,15 @@ module Typ :
     val extension: ?loc:loc -> ?attrs:attrs -> extension -> core_type
 
     val force_poly: core_type -> core_type
+
+    val varify_constructors: str list -> core_type -> core_type
+    (** [varify_constructors newtypes te] is type expression [te], of which
+        any of nullary type constructor [tc] is replaced by type variable of
+        the same name, if [tc]'s name appears in [newtypes].
+        Raise [Syntaxerr.Variable_in_scope] if any type variable inside [te]
+        appears in [newtypes].
+        @since 4.05
+     *)
```

Comment:
May I suggest a slightly different wording?
``` [varify_constructors newtypes te] is the type expression [te'], 
     where all nullary type constructors whose name appears in [newtypes]
     have been replaced by a type variable of the same name.
     Raise [Syntaxerr.Variable_in_scope] if any type variable inside [te]
     appears in [newtypes].
     @since 4.05
     *)
```
N.B: Please feel free to ignore or improve upon this suggestion.